From 511bd4fee7fa0c2abf394681713c6bce93b99b02 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Wilmet?= <swilmet@gnome.org>
Date: Thu, 12 Nov 2015 17:23:16 +0100
Subject: [PATCH 2/2] Revert "inspector: remove
 gtk_text_buffer_get_iter_at_line_index() workaround"

This reverts commit 1536710dbac58f0d46a6c308370e1e3864bf42e0.

Because of the revert at
commit 0dc66f5125196ee10c7e851b75f16e9f5669eb49.
---
 gtk/inspector/css-editor.c | 37 +++++++++++++++++++++++++++++--------
 1 file changed, 29 insertions(+), 8 deletions(-)

diff --git a/gtk/inspector/css-editor.c b/gtk/inspector/css-editor.c
index b11329d..776b3ac 100644
--- a/gtk/inspector/css-editor.c
+++ b/gtk/inspector/css-editor.c
@@ -249,6 +249,27 @@ text_changed (GtkTextBuffer         *buffer,
   ce->priv->timeout = g_timeout_add (100, update_timeout, ce); 
 }
 
+/* Safe version of gtk_text_buffer_get_iter_at_line_index(). */
+static void
+safe_get_iter_at_line_index (GtkTextBuffer *buffer,
+                             GtkTextIter   *iter,
+                             gint           line_number,
+                             gint           byte_index)
+{
+  if (line_number >= gtk_text_buffer_get_line_count (buffer))
+    {
+      gtk_text_buffer_get_end_iter (buffer, iter);
+      return;
+    }
+
+  gtk_text_buffer_get_iter_at_line (buffer, iter, line_number);
+
+  if (byte_index < gtk_text_iter_get_bytes_in_line (iter))
+    gtk_text_iter_set_line_index (iter, byte_index);
+  else
+    gtk_text_iter_forward_to_line_end (iter);
+}
+
 static void
 show_parsing_error (GtkCssProvider        *provider,
                     GtkCssSection         *section,
@@ -259,14 +280,14 @@ show_parsing_error (GtkCssProvider        *provider,
   const char *tag_name;
   GtkTextBuffer *buffer = GTK_TEXT_BUFFER (ce->priv->text);
 
-  gtk_text_buffer_get_iter_at_line_index (buffer,
-                                          &start,
-                                          gtk_css_section_get_start_line (section),
-                                          gtk_css_section_get_start_position (section));
-  gtk_text_buffer_get_iter_at_line_index (buffer,
-                                          &end,
-                                          gtk_css_section_get_end_line (section),
-                                          gtk_css_section_get_end_position (section));
+  safe_get_iter_at_line_index (buffer,
+                               &start,
+                               gtk_css_section_get_start_line (section),
+                               gtk_css_section_get_start_position (section));
+  safe_get_iter_at_line_index (buffer,
+                               &end,
+                               gtk_css_section_get_end_line (section),
+                               gtk_css_section_get_end_position (section));
 
   if (g_error_matches (error, GTK_CSS_PROVIDER_ERROR, GTK_CSS_PROVIDER_ERROR_DEPRECATED))
     tag_name = "warning";
-- 
2.6.2

